### PR TITLE
Insecure backend proxy

### PR DESCRIPTION
**Pull request by deads2k**
KEP https://github.com/kubernetes/enhancements/pull/1261

This adds a podlogoptions value for skipping tls verification of kubelets.

/priority important-soon

```release-note
An end-user may choose to request logs without confirming the identity of the backing kubelet.  This feature can be disabled by setting the `AllowInsecureBackendProxy` feature-gate to false.
```

### CODE REVIEW

**Comment by sttts** [comment]
/retest
Lgtm.

/assign @kubernetes/api-reviewers

File: pkg/kubelet/client/kubelet_client.go [hunk 1]
```diff
@@ -65,20 +65,30 @@ type KubeletClientConfig struct {
...
 // ConnectionInfoGetter provides ConnectionInfo for the kubelet running on a named node
 type ConnectionInfoGetter interface {
 	GetConnectionInfo(ctx context.Context, nodeName types.NodeName) (*ConnectionInfo, error)
 }
 
-// MakeTransport creates a RoundTripper for HTTP Transport.
-func MakeTransport(config *KubeletClientConfig) (http.RoundTripper, error) {
-	tlsConfig, err := transport.TLSConfigFor(config.transportConfig())
+// makeTransport creates a RoundTripper for HTTP Transport.
+func makeTransport(config *KubeletClientConfig, insecureSkipTLSVerify bool) (http.RoundTripper, error) {
```
Comment (smarterclayton) [inline pkg/kubelet/client/kubelet_client.go]: I'd prefer you didn't make this private.

**Review by smarterclayton — CHANGES_REQUESTED** [review]
One comment

File: pkg/kubelet/client/kubelet_client.go (hunk 1 above)
Comment (deads2k) [reply pkg/kubelet/client/kubelet_client.go]: > I'd prefer you didn't make this private.

exposed as two explicit methods.  `MakeTransport` - creates a secure transport as before.  `MakeInsecureTransport` produces an insecure transport.  The diff of diffs is clean: https://github.com/kubernetes/kubernetes/compare/b855d99d58397c4a8704091bf34b7ee7e2458c54..867ee1d5ff88b42261c1224813779a66318d70b2

Comment (smarterclayton) [reply pkg/kubelet/client/kubelet_client.go]: thanks

**Pull request merged**